### PR TITLE
feat(compute): Add Zonal MIG example

### PR DESCRIPTION
**Pull request by askubis**
## Description

Fixes #216660865

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [X] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [X] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [X] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [X] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [X] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [X] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s): https://cloud.google.com/compute/docs/instance-groups/create-zonal-mig

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [X] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [X] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

File: compute/zonal_instance_group_manager/main.tf [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# [START compute_zonal_instance_group_manager_parent_tag]
+data "google_compute_image" "debian_11" {
+
```
Comment (msampathkumar) [inline compute/zonal_instance_group_manager/main.tf:19]:
```suggestion
```

File: compute/zonal_instance_group_manager/main.tf [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# [START compute_zonal_instance_group_manager_parent_tag]
+data "google_compute_image" "debian_11" {
```
Comment (msampathkumar) [inline compute/zonal_instance_group_manager/main.tf:18]:
```suggestion
data "google_compute_image" "default" {
```

As a common practice use `default`, unless you have a name conflict.

File: compute/zonal_instance_group_manager/main.tf [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  name = "my-igm"
+  zone = "us-central1-f"
+
+  version {
+    instance_template = google_compute_instance_template.default.id
+    name              = "primary"
+  }
+
+  base_instance_name = "autoscaler-sample"
```
Comment (msampathkumar) [inline compute/zonal_instance_group_manager/main.tf:58]: Just suggestion. You can discard it. All other resources were given a prefix of `my-`, so changing this to `my-igm-autoscalar` or something similar could be a good pattern.
```suggestion
  base_instance_name = "my-igm-autoscaler"
```

**Comment by msampathkumar** [comment]
/gcbrun

Invoking CICD tests.

File: compute/zonal_instance_group_manager/main.tf (hunk 3 above)
Comment (askubis) [reply compute/zonal_instance_group_manager/main.tf]: fixed it differently, using my-igms-instance as the base name.

File: compute/zonal_instance_group_manager/main.tf [hunk 4]
```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2022 Google LLC
```
Comment (iennae) [inline compute/zonal_instance_group_manager/main.tf:2]: 2023 since this is a new file

**Review by iennae — APPROVED** [review]
needs the copyright to reflect that this was made this year and not copypastaed from an existing sample.

**Pull request merged**